File: priklady/java/david/matematickepriklady/models/Task.java
package david.matematickepriklady.models;

import david.matematickepriklady.ui.activities.ChooseActivity;

public class Task {

    private int a, b, answer;
    private int exerciseType;
    private int roof;
    private int limit = 0;
    private String strAns;

    public Task(){

    }

    public void prepare(){

        switch (exerciseType){
            case ChooseActivity.ADDITION:
                this.answer = this.a + this.b;
                break;
            case ChooseActivity.SUBSTRACTION:
                this.answer = this.a - this.b;
                break;
            case ChooseActivity.MULTIPLICATION:
                this.answer = this.a * this.b;
                break;
            case ChooseActivity.DIVISION:
                this.answer = this.a / this.b;
                break;
            case ChooseActivity.COMPARISION:
                if(a > b) strAns = " > ";
                else if(a < b) strAns = " < ";
                else if (a == b) strAns = " = ";
                break;
        }
    }


    @Override
    public String toString(){

        switch (exerciseType){
            case ChooseActivity.ADDITION:
                return ""+ a +" + "+ b +" = ";
            case ChooseActivity.SUBSTRACTION:
                return ""+ a +" - "+ b +" = ";
            case ChooseActivity.MULTIPLICATION:
                return ""+ a +" x "+ b +" = ";
            case ChooseActivity.DIVISION:
                return ""+ a +" / "+ b +" = ";
            case ChooseActivity.COMPARISION:
                return ""+ a + "  ?  " + b+" = ";
        }

        return "";
    }

    public int getA() {
        return a;
    }

    public int getB() {
        return b;
    }

    public int getAnswer() {
        return answer;
    }

    public int getExerciseType() {
        return exerciseType;
    }

    public void setA(int a) {
        this.a = a;
    }

    public void setB(int b) {
        this.b = b;
    }

    public void setAnswer(int answer) {
        this.answer = answer;
    }

    public void setExerciseType(int exerciseType) {
        this.exerciseType = exerciseType;
    }

    public int getRoof() {
        return roof;
    }

    public void setRoof(int roof) {
        this.roof = roof;
    }

    public int getLimit() {
        return limit;
    }

    public void setLimit(int limit) {
        this.limit = limit;
    }


}

File: priklady/java/david/matematickepriklady/models/Task.kt
package com.dejvid.matematika.models

import android.provider.ContactsContract

class Task : ITask{


    private var id = 0
    private var a = 0
    private var b = 0
    private var exerciseType = 0
    private var difficulty = 0
    private var limit = 0
    private var wrongAnswers : IntArray? = null
    private var result : String = ""


    fun setID(id : Int){
        this.id = id
    }

    fun getID() : Int {
        return this.id
    }

    override fun setA(a : Int){
        this.a = a
    }

    override fun getA() : Int{
        return this.a
    }

    override fun setB(b : Int){
        this.b = b
    }

    override fun getB() : Int{
        return this.b
    }

    override fun setExerciseType(type: Int) {
        this.exerciseType = type
    }

    override fun getExerciseType(): Int {
        return this.exerciseType
    }

    override fun setDifficulty(diff: Int) {
        this.difficulty = diff
    }

    override fun getDifficulty(): Int {
       return this.difficulty
    }

    override fun setLimit(limit: Int) {
        this.limit = limit
    }

    override fun getLimit(): Int {
        return this.limit
    }

    override fun setResult(res: String) {
        this.result = res
    }

    override fun getResult() : String{
        return this.result
    }

    fun getWrongAnswers(): IntArray? {
        if(this.wrongAnswers != null){
            return this.wrongAnswers
        } else {
            return null
        }
    }

    fun setWrongAnswers(wa : IntArray){
        this.wrongAnswers = wa
    }

    override fun toString(): String {
        super.toString()
        var result = this.a.toString()

        when(this.exerciseType) {
            Constants.ADDITION.value -> result += " + "
            Constants.SUBTRACTION.value -> result += " - "
            Constants.MULTIPLICATION.value -> result += " * "
            Constants.DIVISION.value -> result += " : "
            Constants.COMPARISON.value -> result += "  ?  "
        }

        result += this.b.toString()

        if(this.exerciseType != Constants.COMPARISON.value) { result += " = " }

        return result
    }

}